unblob/handlers/compression/_gzip_reader.py

```diff
@@ -15,7 +15,7 @@ def _add_read_data(self, data):
         self._crc = zlib.crc32(data, self._crc)
         self._stream_size = self._stream_size + len(data)
 
-    def read(self):
+    def read_all(self):
```

Comment:
There are more serious potential problems here than a linter problem: this gzip reader is an override of a private third-party class, whose implementation might change any moment.

So I think it is unjustified and somewhat dangerous to change the name of a known to work function (due to a linter problem).
I would just add a type ignore comment here for the original code.

Reply:
> gzip reader is an override of a private third-party class, whose implementation might change any moment.

That's why I thought it is more dangerous to override it than to just extend it with another function.

If in the future any part of this base class that we use would call `read()`, it would either behave incorrectly or explode in case if it would pass in an argument.

Reply:
Yes, I agree with your logic. Yet I do not like changing code which works, because a linter says so.
Also, I think we do not run pyright in a matrix like the tests, so I think we would not catch problems as easily, as described, and thus we could also hide a newly popping up function accidentally.

Reply:
I did say the exact opposite of that: it would be hard to catch this kind of bug. My assessment is that renaming this function reduces that risk.
The change is not because of a linter warning, but because breaking the parent class's behavior is not a good idea in the first place.

Reply:
I like the fix as it is.